Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift

```diff
@@ -0,0 +1,136 @@
+import Foundation
+import Services
+import Combine
+import AnytypeCore
+
+protocol SubscriptionStorageProtocol: AnyObject {
+    var subId: String { get }
+    var detailsStorage: ObjectDetailsStorage { get }
+    
+    func startOrUpdateSubscription(data: SubscriptionData, update: @MainActor @escaping (_ state: SubscriptionStorageState) -> Void) async throws
+    func stopSubscription() async throws
+}
+
+actor SubscriptionStorage: SubscriptionStorageProtocol {
+    
+    // MARK: - DI
+    
+    nonisolated let subId: String
+    nonisolated let detailsStorage: ObjectDetailsStorage
+    private let toggler: SubscriptionTogglerProtocol
+    
+    // MARK: - State
+    
+    private var subscription: AnyCancellable?
+    
+    private var data: SubscriptionData?
+    private var update: (@MainActor (_ data: SubscriptionStorageState) -> Void)?
+    
+    private var orderIds: [BlockId] = []
+    private var state = SubscriptionStorageState(total: 0, nextCount: 0, prevCount: 0, items: [])
+    
+    init(subId: String, detailsStorage: ObjectDetailsStorage, toggler: SubscriptionTogglerProtocol) {
+        self.subId = subId
+        self.detailsStorage = detailsStorage
+        self.toggler = toggler
+        subscription = EventBunchSubscribtion.default.addHandler { [weak self] events in
+            guard events.contextId.isEmpty else { return }
+            await self?.handle(events: events)
+        }
+    }
+    
+    deinit {
+        guard let data else { return }
+        Task { [data, toggler] in
+            try await toggler.stopSubscription(id: data.identifier)
+        }
```

Comment:
you can call `stopSubscription()` here

Reply:
This is deinit. And if i call `stopSubscription` there will be a crash

Reply:
self shouldn't be capture

Reply:
didn’t notice, agree!